File: Data Structures & Algorithms/majority-element/submission-3.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
       hm = {}
        
       for n in nums:
        if n not in hm:
            hm[n] = 1
        else:
            hm[n]+=1
       
       m = nums[0]
       for k in hm:
        if hm[k] == hm[m]:
            if k > m:
                m = k
        if hm[k] > hm[m]:
                m = k
       return m

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++) {
            int prod = 1;
            for(int j = 0; j < nums.length; j++) {
                if(i != j) prod *= nums[j]; 
            }
            map.put(nums[i], prod);
        }
        int[] res = new int[nums.length];
        for(int i = 0; i < res.length; i++) {
            res[i] = map.get(nums[i]);
        }
        return res;
     }
}  
